### ISSUE TITLE
Spike: Implications of having just one combined submission video (goal W 9/11)  [20,13,1.54]

### ISSUE THREAD

**Issue by hellafitz** [Iridescent-CM/technovation-app#2177]
Submission changes need to be communicated to external partners, so before we decide on a go/no-go Program team has asked for dev to investigate the implications of combined the pitch and demo videos into one, unified video for submission. 

This ticket would ideally be completed by Friday 9/6. It is the highest priority of the two potential submission changes. 

Tasks:
- Investigate the platform ramifications of this change
- scope the work required to implement 
- communicate to Product Team the findings, so Devin can be informed asap

Relevant 8/28 TG Product meeting notes https://docs.google.com/document/d/14Dx9FCS6UUn6g_nMh1bBJOO998AOqM7fET83s2hXUjo/edit#bookmark=id.gflrgn2wwrk0

"TGirls SUBMISSION CHANGES
1 video (combine pitch + demo, create very clear guidelines on what goes into each)
- Explore how to have 1 video submitted
- Need to think through how this works with historical data

Objectives
reduce the load on the participants for their submission (also, additional video doesn’t add to learning experience)
Indicators: more completed submissions (but that’s a lot of factors), benchmark Q: what % of incompleted submissions from prev year only have one video link?
Reduce staff admin time managing two correct links for every submission
Benchmark data: spreadsheet of how many teams we had to correct for video link issues
Most common issue currently: they don’t make the video public
2nd: kids paste in the same video link twice
Indicator: reduced # of team video link issues"

**Comment by hellafitz** [comment]
Amy Q: does this include the Judging view? A: yes - focussed on could we do it, not exactly how.

**Comment by stenington** [comment]
## Platform ramifications

The major issue is that the platform can't handle different seasons having different submission requirements, so this change will corrupt our past season submission data. I think the basic options for dealing with that are:

* leave the slightly corrupted data there and hope nobody gets confused by it
* delete old submissions but leave old team data, and hope nobody gets confused thinking a team that did previously submit didn't
* delete old submission and team data, and essentially "start fresh" this coming season

The platform mostly deals with the "current season" whichever season that is, so hopefully there wouldn't be huge ramifications, but I can't chase down everywhere that the platform, related tools, dataclips, etc. might look at past seasons to know 100% what will happen.

## Scope

Before I can scope the work required, I need to know specifically what the changes are. Here are the places I found that the demo and pitch videos appear on the platform:

**Admin/RA**
![Screen Shot 2019-09-05 at 10 26 03 AM](https://user-images.githubusercontent.com/55319/64355252-e133ed00-cfce-11e9-81e6-8350056aef11.png)
![Screen Shot 2019-09-05 at 10 44 49 AM](https://user-images.githubusercontent.com/55319/64355264-e2651a00-cfce-11e9-9b5a-00642b81dd09.png)


**Student and Mentor**
![Screen Shot 2019-09-05 at 10 30 05 AM](https://user-images.githubusercontent.com/55319/64355253-e1cc8380-cfce-11e9-88e4-f6e980c41bc2.png)
![Screen Shot 2019-09-05 at 10 30 13 AM](https://user-images.githubusercontent.com/55319/64355255-e1cc8380-cfce-11e9-9f1d-90bbe4ff18b1.png)
![Screen Shot 2019-09-05 at 10 30 17 AM](https://user-images.githubusercontent.com/55319/64355257-e1cc8380-cfce-11e9-8efe-afd9d697e0dd.png)
![Screen Shot 2019-09-05 at 10 30 46 AM](https://user-images.githubusercontent.com/55319/64355258-e1cc8380-cfce-11e9-83ae-77b780c41406.png)
![Screen Shot 2019-09-05 at 10 41 30 AM](https://user-images.githubusercontent.com/55319/64355259-e1cc8380-cfce-11e9-8fc0-d8931587ec3f.png)

**Judge**

![Screen Shot 2019-09-05 at 10 42 18 AM](https://user-images.githubusercontent.com/55319/64355261-e1cc8380-cfce-11e9-8916-fd3b1d8331b5.png)
![Screen Shot 2019-09-05 at 10 42 31 AM](https://user-images.githubusercontent.com/55319/64355262-e2651a00-cfce-11e9-9c02-9d0da222f901.png)

My intuition is that anywhere there are two separate items, we'd replace them with a single version and call it "Demo and pitch video", or something like that? 

Additionally, it's worth noting that part of the scoring rubric is internally labelled "demo video":

![Screen Shot 2019-09-05 at 11 04 22 AM](https://user-images.githubusercontent.com/55319/64355694-a7171b00-cfcf-11e9-8ada-b63b5fa5b51b.png)

We might consider renaming that to "usability" or something along those lines to avoid confusion, assuming the rubric doesn't otherwise change.

**Comment by stenington** [comment]
Amy pointed out that admins do access previous year data to see if a current year submission is different or not from their previous submissions, so that would be an issue for removing old data. Just flagging, not yet deciding.

**Comment by stenington** [comment]
Also flagging that we should take a long-lived backup before making these changes.

**Comment by stenington** [comment]
Also noting that RA's do have the "season" filter on both the teams and submissions sections of their admin screens, although Amy reports that many RAs just email asking for spreadsheets. But the first decision point would affect our admins **and RA accounts**.

**Comment by stenington** [comment]
There's interest in option 2: drop old submissions but not old teams. It's worth digging a little deeper into what this would look like.

**Comment by stenington** [comment]
Double check on the public submissions page

**Comment by stenington** [comment]
Public submission page uses the `team_submissions/published` template. Videos are not embedded on it, they're still the links that open modals.

**Comment by stenington** [comment]
Other tables referencing `team_submission_id`:
* `business_plans`
* `pitch_presentations`
* `screenshots`
* `submission_scores`
* `technical_checklists`

So between this ticket and #2178 we'd be talking about removing submissions (including business plans, pitch presentations, screenshots, and technical checklists) and removing past scores as well.

I think it's probably appropriate to open a new ticket for dealing with the plan for archiving what we think we will need.

**Comment by stenington** [comment]
In terms of closing out this spike, I've created #2187 and #2191 which both contain work required to make this change. We needed to quickly give a go/no-go decision on this one because the requirement changes need to be communicated in the curriculum, but the actual platform work doesn't need to be completed until submissions open in early January. 

I feel this ticket is a go, with the caveat that once we commit to the change, the ticket needs to be prioritized accordingly so the work can happen in a timely fashion. Despite the late deadline, there's no reason I see not to begin work on this change much more quickly.
